sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2210,25 +2216,51 @@ def update_ssh_ports(self, max_attempts: int = 1) -> None:
         Use this method to use any cloud-specific port fetching logic.
         """
         del max_attempts  # Unused.
-        if isinstance(self.launched_resources.cloud, clouds.RunPod):
-            cluster_info = provision_lib.get_cluster_info(
-                str(self.launched_resources.cloud).lower(),
-                region=self.launched_resources.region,
-                cluster_name_on_cloud=self.cluster_name_on_cloud,
-                provider_config=None)
-            self.stable_ssh_ports = cluster_info.get_ssh_ports()
+        if self.cached_cluster_info is not None:
+            self.stable_ssh_ports = self.cached_cluster_info.get_ssh_ports()
```

Comment:
Why do we need to update ssh ports for all cloud now while previously we only do it for runpod?

Reply:
This is because we did not cache the `cluster_info` for other clouds, so we did not fetch the ports from `cluster_info` for optimization. Now, we have the cluster_info cached, we can use the ssh_ports stored in cluster_info for generality.